### PR TITLE
Added is_floating_point for Paddle frontend

### LINKED ISSUE #17970: is_floating_point

### PR DESCRIPTION
**Pull request by arshPratap**
Close #17970

### CODE REVIEW

**Comment by arshPratap** [comment]
@hmahmood24 please take a look thanks !

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Comment by arshPratap** [comment]
add_frontend_checklist

**Comment by arshPratap** [comment]
@theRealBird @hmahmood24 I have rebased the current branch with the latest patches in master and have also updated the task list to the best of my knowledge ...please take another look at this. Thanks a lot !

**Comment by theRealBird** [comment]
Hey @arshPratap,
The PR looks good to me! Thanks a lot for contributing to Ivy. 
I do have a request. Before we merge this, could you please resolve the merge conflicts? Thanks!

**Comment by arshPratap** [comment]
> Hey @arshPratap, The PR looks good to me! Thanks a lot for contributing to Ivy. I do have a request. Before we merge this, could you please resolve the merge conflicts? Thanks!

Done ! @theRealBird thanks for your review. I have rebased the branch with `master`. PTAL .Thanks again !

**Comment by arshPratap** [comment]
@theRealBird have rebased it to the latest changes on  `master` branch . Can this be merged now

**Comment by arshPratap** [comment]
@theRealBird sorry for the delay ! Have rebased to the latest commits ! Please take another look. Thanks

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_tensor.py
Comment (theRealBird) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_tensor.py:1]: @arshPratap, Thanks for making the changes. I was going through the code once more and I think passing only "float" data-types to a function which checks if a value is a float or not is not the correct approach for testing. We are not checking how does the function behave when the value passed is not a float.

**Review by theRealBird — CHANGES_REQUESTED** [review]

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_tensor.py
Comment (arshPratap) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_tensor.py:1]: Yup yup..completely true..lemme add in a quick fix for this

Comment (arshPratap) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_tensor.py:1]: @theRealBird could you take a look now ..decided to move with the `numeric` data types as compared to the previous `float` data type

**Comment by arshPratap** [comment]
@theRealBird have made the changes you requested .. would love to hearback on any thoughts you have regarding this PR.Thanks

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by arshPratap** [comment]
@theRealBird So sorry for the delay..I have rebased the PR to the current changes in the master..Please take a look.Thanks again

**Comment by theRealBird** [comment]
Hey @arshPratap, I am sorry for the delay at my end too. 
Not sure how this skipped under my radar. Could you please take care of the conflicts one last time (hopefully)?
I will be careful to merge it in time. Thanks!

**Review by theRealBird — CHANGES_REQUESTED** [review]
Just need to resolve the merge conflicts. Thanks!

**Comment by arshPratap** [comment]
@theRealBird done !!

**Review by theRealBird — CHANGES_REQUESTED** [review]
Could you please verify if the test you have implemented is passing? Currently it appears to be failing on the CI dashboard. 

Thanks!

**Comment by arshPratap** [comment]
@theRealBird thanks for the .. review and so sorry for the late reply..lemme work on failing tests and I will update the thread here regarding the same.
Thanks

**Comment by arshPratap** [comment]
@theRealBird fixed the tests..seems that there are some datatypes that are not compatible with paddle (unsigned integers being the case) ..i had to remove the `"valid"` and `"numeric"` data types and replace them

**Comment by arshPratap** [comment]
@theRealBird ...the tests have been completed ..seems like the is_floating_point tests are not failing anymore

**Review by theRealBird — APPROVED** [review]
LGTM

**Pull request merged**